yt_dlp/extractor/tver.py

```diff
@@ -3,75 +3,40 @@
     traverse_obj,
 )
 
 
 class TVerIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?tver\.jp/(?P<path>corner|episode|feature|lp|tokyo2020/video)/(?P<id>[fc]?\d+)'
+    _VALID_URL = r'https?://(?:www\.)?tver\.jp/(?P<path>corner|episodes?|feature|lp|tokyo2020/video)/(?P<id>[a-zA-Z0-9]+)'
     # videos are only available for 7 days
     _TESTS = [{
-        'url': 'https://tver.jp/corner/f0062178',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/feature/f0062413',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/episode/79622438',
-        'only_matching': True,
-    }, {
-        # subtitle = ' '
-        'url': 'https://tver.jp/corner/f0068870',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/lp/f0009694',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/lp/c0000239',
-        'only_matching': True,
-    }, {
-        'url': 'https://tver.jp/tokyo2020/video/6264525510001',
+        'url': 'https://tver.jp/episodes/ephss8yveb',
         'only_matching': True,
     }]
-    _TOKEN = None
     BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/%s/default_default/index.html?videoId=%s'
 
-    def _real_initialize(self):
-        self._TOKEN = self._download_json(
-            'https://tver.jp/api/access_token.php', None)['token']
-
     def _real_extract(self, url):
-        path, video_id = self._match_valid_url(url).groups()
-        if path == 'lp':
-            webpage = self._download_webpage(url, video_id)
-            redirect_path = self._search_regex(r'to_href="([^"]+)', webpage, 'redirect path')
-            path, video_id = self._match_valid_url(f'https://tver.jp{redirect_path}').groups()
-        api_response = self._download_json(f'https://api.tver.jp/v4/{path}/{video_id}', video_id, query={'token': self._TOKEN})
-        p_id = traverse_obj(api_response, ('main', 'publisher_id'))
-        if not p_id:
-            error_msg, expected = traverse_obj(api_response, ('episode', 0, 'textbar', 0, ('text', 'longer')), get_all=False), True
-            if not error_msg:
-                error_msg, expected = 'Failed to extract publisher ID', False
-            raise ExtractorError(error_msg, expected=expected)
-        service = remove_start(traverse_obj(api_response, ('main', 'service')), 'ts_')
-
-        r_id = traverse_obj(api_response, ('main', 'reference_id'))
-        if service not in ('tx', 'russia2018', 'sebare2018live', 'gorin'):
-            r_id = 'ref:' + r_id
+        video_id = self._match_id(url)
+        video_info = self._download_json(
+            f'https://statics.tver.jp/content/episode/{video_id}.json', video_id,
+            query={'v': '5'}, headers={
+                'Origin': 'https://tver.jp',
+                'Referer': 'https://tver.jp/',
+            })
+        p_id = traverse_obj(video_info, ('video', 'accountID'))
+        r_id = 'ref:' + traverse_obj(video_info, ('video', 'videoRefID'))
```

Comment:
With https://tver.jp/lp/episodes/ep6f16g26p, there's no `videoRefID`:
```
{
...
 'video': {'accountID': '3971130137001',
           'channelID': 'tx',
           'playerID': 'Eyc2R2Jow',
           'videoID': '6302378806001'},
 ...
}
```
Apparently `videoID` isn't a good substitute:
```
[brightcove:new] ref:6302378806001: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found ...
```

Reply:
Ah I wasn't aware of teasers :sweat_smile: